tritium-metrics/src/test/java/com/palantir/tritium/metrics/MetricRegistriesTest.java

```diff
@@ -394,6 +396,40 @@ public void problematicTimestamps() {
         assertParsesTimestamp("2019-03-29T23:38:51.092Z");
     }
 
+    @Test
+    public void codahale_registry_histogram_count_should_monotonically_increase_after_window() throws InterruptedException {
+        MetricRegistry registry =
+                MetricRegistries.createWithSlidingTimeWindowReservoirs(1, TimeUnit.MILLISECONDS);
+
+        Histogram histogram = registry.histogram("histogram");
+        histogram.update(20);
+        histogram.update(20);
+        histogram.update(20);
+        histogram.update(20);
+
+        assertThat(histogram.getCount()).isEqualTo(4);
+
+        Thread.sleep(1);
+
+        assertThat(histogram.getCount()).isEqualTo(4); // wat
```

Comment:
This fails because this counter returns 0 but the tagged counter returns 4 🤯

Reply:
Hmmm seems like we knew about this originally @kasper0406 - was this intentional? https://github.com/palantir/tritium/pull/238/files#diff-52a48109605a97396b162e386757ef08R154

Reply:
Not intentional in the sense of any of my use cases. As noted below, this seems to be an implication of the sliding window, where the count is counting only the entries constituting the reported metric.
I do see that this could give unintended behavior if used with fx the `rate` function in Prometheus, as the count will now be sliding. Didn’t consider that. Seems better if it reported two counts - one being the current count, and one the total. Don’t know if that’s  possible though.